data/transactions/logic/README.md

```diff
@@ -40,6 +40,32 @@ has fewer than two elements, the operation fails. Some operations, like
 `frame_dig` and `proto` could fail because of an attempt to access
 above the current stack.
 
+## Stack Types
+
+While every element of the stack is restricted to the types `uint64` and `bytes`, 
+the values of these types may be known to be bounded.  The more common bounded types are 
+named to provide more semantic information in the documentation. They're also used during
+assembly time to do type checking and to provide more informative error messages.
+
+
+#### Definitions
+
+| Name | Bound | AVM Type |
+| ---- | ---- | -------- |
+| uint64 | x <= 18446744073709551615 | uint64 |
+| none |  | none |
+| name | 1 <= len(x) <= 64 | []byte |
+| method | len(x) == 4 | []byte |
+| key | len(x) <= 64 | []byte |
+| bool | x <= 1 | uint64 |
+| bigint | len(x) <= 64 | []byte |
+| any |  | any |
+| address | len(x) == 32 | []byte |
+| []byte | len(x) <= 4096 | []byte |
+| [32]byte | len(x) == 32 | []byte |
```

Comment:
Should we remove the `[32]byte` line and just hard code a line like:
`| [N]byte | len(x) == N | []byte |`

I'd probably remove the `method` time and use `[4]byte` directly if we did that.

Just looking for thoughts, no need to make changes here. I want to get this in.